src/test/kotlin/net/atos/zac/signalering/SignaleringServiceTest.kt

```diff
@@ -0,0 +1,86 @@
+import io.mockk.checkUnnecessaryStub
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import jakarta.persistence.EntityManager
+import jakarta.persistence.Query
+import jakarta.persistence.criteria.CriteriaBuilder
+import jakarta.persistence.criteria.CriteriaDelete
+import jakarta.persistence.criteria.Path
+import jakarta.persistence.criteria.Predicate
+import jakarta.persistence.criteria.Root
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.zac.app.zaken.converter.RESTZaakOverzichtConverter
+import net.atos.zac.event.EventingService
+import net.atos.zac.mail.MailService
+import net.atos.zac.signalering.model.Signalering
+import java.time.ZonedDateTime
+
+class SignaleringServiceTest : BehaviorSpec({
+    val eventingService = mockk<EventingService>()
+    val mailService = mockk<MailService>()
+    val signaleringenMailHelper = mockk<SignaleringMailHelper>()
+    val signaleringZACHelper = mockk<SignaleringZACHelper>()
+    val signaleringPredicateHelper = mockk<SignaleringPredicateHelper>()
+    val zrcClientService = mockk<ZRCClientService>()
+    val restZaakOverzichtConverter = mockk<RESTZaakOverzichtConverter>()
+    val entityManager = mockk<EntityManager>()
+    val criteriaBuilder = mockk<CriteriaBuilder>()
+    val criteriaDeleteSignalering = mockk<CriteriaDelete<Signalering>>()
+    val rootSignalering = mockk<Root<Signalering>>()
+    val pathTijdstip = mockk<Path<Long>>()
+    val query = mockk<Query>()
+
+    val signaleringService = SignaleringService(
+        eventingService = eventingService,
+        mailService = mailService,
+        signaleringenMailHelper = signaleringenMailHelper,
+        signaleringZACHelper = signaleringZACHelper,
+        signaleringPredicateHelper = signaleringPredicateHelper,
+        zrcClientService = zrcClientService,
+        restZaakOverzichtConverter = restZaakOverzichtConverter
+    )
+    signaleringService.entityManager = entityManager
+
+    beforeEach {
+        checkUnnecessaryStub()
+    }
+
+    beforeSpec {
+        clearAllMocks()
+    }
+
+    Given("signaleringen older than two days") {
+        val now = ZonedDateTime.now()
+        val deleteOlderThanDays = 2L
+        val zoneDateTimeSlot = slot<ZonedDateTime>()
+        every { entityManager.criteriaBuilder } returns criteriaBuilder
+        every { criteriaBuilder.createCriteriaDelete(Signalering::class.java) } returns criteriaDeleteSignalering
+        every { criteriaDeleteSignalering.from(Signalering::class.java) } returns rootSignalering
+        every { rootSignalering.get<Long>("tijdstip") } returns pathTijdstip
+        every { criteriaBuilder.lessThan(any(), capture(zoneDateTimeSlot)) } returns mockk<Predicate>()
+        every { criteriaDeleteSignalering.where(any<Predicate>()) } returns criteriaDeleteSignalering
+        every { entityManager.createQuery(criteriaDeleteSignalering) } returns query
+        every { query.executeUpdate() } returns 2
```

Comment:
Everything to do with the actual delete is mocked, so what logic is really tested here?

The only logic that is not mocked is `ZonedDateTime.now().minusDays(deleteOlderThanDays)`, and those are also library calls.

So even testing for that seems a bit much.... but, 🆗

Reply:
Yes true, I also noticed this when I started writing the test. However all those mocks to some degree do act as the unit test themselves as well I think. I mean: you need to mock exactly the right services and methods because if you don't the test will fail. In that way a test like this does test if the code does what you expect it to somewhat..

Reply:
Or in other words: there is hardly any business logic here to test, but the most important business logic is tested I think. But yeah, is all this effort really worth it or not is another discussion..